libraries/tlv-account-resolution/src/state.rs

```diff
@@ -105,6 +143,35 @@ impl ExtraAccountMetaList {
             .saturating_add(PodSlice::<ExtraAccountMeta>::size_of(num_items)?))
     }
 
+    /// Checks provided account infos against validation data, using
+    /// instruction data and program ID to resolve any dynamic PDAs
+    /// if necessary.
+    pub fn check_account_infos<T: SplDiscriminate>(
+        account_infos: &[AccountInfo],
+        instruction_data: &[u8],
+        program_id: &Pubkey,
+        data: &[u8],
+    ) -> Result<(), ProgramError> {
+        let state = TlvStateBorrowed::unpack(data).unwrap();
+        let extra_meta_list = ExtraAccountMetaList::unpack_with_tlv_state::<T>(&state)?;
+        let extra_account_metas = extra_meta_list.data();
+
+        let provided_metas = account_infos
+            .iter()
+            .map(account_meta_from_info)
+            .collect::<Vec<_>>();
+
+        for config in extra_account_metas.iter() {
+            let meta = config.resolve(&provided_metas, instruction_data, program_id)?;
+            provided_metas
+                .iter()
+                .find(|&x| *x == meta)
+                .ok_or(AccountResolutionError::IncorrectAccount)?;
+        }
```

Comment:
Is it also possible to ensure the order is correct? I think if you pass in `account_info_iter.as_slice()` you should also be able to enumerate and make sure the order is respected. And can you add a test that correctly errors if they're provided out of order?

Reply:
Sure thing!